app/services/repository_datatable_service.rb

```diff
@@ -433,21 +433,23 @@ def sort_rows(column_obj, records)
       return records unless sorting_column
 
       sorting_data_type = sorting_column.data_type.constantize
-      cells = sorting_data_type.joins(:repository_cell)
-                               .where('repository_cells.repository_column_id': sorting_column.id)
+      cells = sorting_data_type.joins(
+        "INNER JOIN repository_cells AS repository_sort_cells "\
+        "ON repository_sort_cells.value_id = #{sorting_data_type.table_name}.id"
```

Comment:
May be it is better to add value_type to join condition for better performance?

Reply:
Added to condition, now works with index.